src/components/Form/InputWrapper.tsx

```diff
@@ -18,11 +18,13 @@ function computeComponentSpecificRegistrationParams<TInput extends ValidInputs>(
     readonly shouldSetTouchedOnBlurOnly: boolean,
 } {
     if (InputComponent === TextInput) {
-        const isEffectivelyMultiline = Boolean(multiline ?? autoGrowHeight);
+        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+        const isEffectivelyMultiline = Boolean(multiline || autoGrowHeight);
 
         // We calculate the effective requested value of `shouldSubmitForm`, assuming that the default value should be
         // `true` for single-line inputs and `false` for multi-line inputs.
-        const shouldSubmitFormOrDefault = shouldSubmitForm ?? !isEffectivelyMultiline;
+        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+        const shouldSubmitFormOrDefault = shouldSubmitForm || !isEffectivelyMultiline;
```

Comment:
Why is this a fix? What is your intention here? As I see it, it makes it impossible to request `shouldSubmitForm = false` for single line inputs.

Reply:
And that should be the case for single line input as per `FORMS.md`.

Reply:
Hmm... So maybe we could nuke `shouldSubmitFormOrDefault` and `canInputSubmitForm`, and go with a single expression in such a case.

```ts
const shouldReallySubmitForm = isEffectivelyMultiline ? Boolean(shouldSubmitForm) && canUseHardwareKeyboard ? true;
```

`shouldReallySubmitForm` could get this comment:

> We want to avoid a situation when the user can't insert a new line. For single-line inputs, it's not a problem and we force-enable form submission. For multi-line inputs, ensure that it was requested to enable form submission for this specific input and that alternative ways exist to add a new line.

My intentions were good, but I might have made it unnecessarily complex.

Reply:
Done.